### ISSUE TITLE
app with babel-node keeps restarting

### ISSUE THREAD

**Issue by spypsy** [Unitech/PM2#1559]
Hi, I'm using pm2 with babel-node interpreter and I find it impossible to kill the app once it has started. I first noticed that reloading after code changes didn't work and logs showed 
```
PM2 Change detected for app name: insights-dev - restarting
PM2 Stopping app:insights-dev id:0
PM2 App name:insights-dev id:0 exited with code SIGTERM
PM2 Process with pid 17912 killed
PM2 Starting execution sequence in -fork mode- for app name:insights-dev id:0
PM2 App name:insights-dev id:0 online
insights-dev-0 Port 3000 is already in use
```
in repeat.

After checking `pm2 ls`, I see this:
```
┌──────────────┬────┬──────┬───────┬────────┬─────────┬────────┬─────────────┬──────────┐
│ App name     │ id │ mode │ pid   │ status │ restart │ uptime │ memory      │ watching │
├──────────────┼────┼──────┼───────┼────────┼─────────┼────────┼─────────────┼──────────┤
│ insights-dev │ 0  │ fork │ 18495 │ online │ 26      │ 2s     │ 18.605 MB   │  enabled │
└──────────────┴────┴──────┴───────┴────────┴─────────┴────────┴─────────────┴──────────┘
```
and the restart number keeps going up.

Even if I run `pm2 delete all` and `pm2 kill`, if I then try to restart the app, I'll have the same problem and I can see the process running with `ps aux | grep babel-node'
```
17938   0.0  1.1  3179668 180928   ??  S     8:42pm   0:04.44 node /usr/local/lib/node_modules/babel/lib/_babel-node
```
I think this might be related to [#1444](https://github.com/Unitech/PM2/issues/1444) but the fix does not work for me. I've tried with pm2 installed locally, globally, both and the same for babel.

Any ideas? What more info could I provide to help you identify the problem?

Edit:

To clarify, I'm running pm2 from a dev.json file, with only options being: 
```
"autorestart": true,
"exec_interpreter": "babel-node",
```

**Comment by soyuka** [comment]
Have you tried `--next-gen-js`? I think that babel is used with this flag (http://pm2.keymetrics.io/docs/usage/specifics/#run-next-generation-javascript).

**Comment by spypsy** [comment]
Thanks for the suggestion. I tried it though, I don't see any difference

**Comment by spypsy** [comment]
Update: I noticed someone in my team who was not experiencing the sam problem had a notice for his PM2 version: 
```
>>>> In-memory PM2 is out-of-date, do:
>>>> $ pm2 updatePM2
In memory PM2 version: 0.12.8
Local PM2 version: 0.14.7
```
I did `npm i pm2@0.12.18` and now the problem seems to be solved!

**Comment by soyuka** [comment]
Interesting, please leave this issue open I'll take a look at it this week-end.

**Comment by soyuka** [comment]
Everything is running fine here by using `pm2@0.14.7` with the `--next-gen-js` flag (nodejs 12.7)

```
>>>> In-memory PM2 is out-of-date, do:
>>>> $ pm2 updatePM2
``` 

In this case you should update the in-memory pm2 by running `pm2 updatePM2`. 

FIY, `--next-gen-js` flag is available since `pm2@0.12.14` and #1444 was fixed in `pm@0.14.4`.
